### PR TITLE
Port Tracr demo to TransformerBridge

### LINKED ISSUE #481: [Bug Report] Demo for Tracr to TransformerLens is kind of broken
**Describe the bug**
[The demo](https://github.com/neelnanda-io/TransformerLens/blob/main/demos/Tracr_to_Transformer_Lens_Demo.ipynb) wrongly assumes that: `The unembed is just a projection onto the first few elements of the residual stream, these store output tokens`. But Tracr allows RASP programs to have named variables, and then it re-orders the residual stream based on the alphabetical order of these names.

The current implementation of the demo works because the last instruction in the RASP program ends in `Aggregate`, which later is automatically assigned the variable name `aggregate_1`, and since it starts with "a", the result gets stored in the first 3 components of the residual stream. See the program used in the demo:

```python
def make_length():
  all_true_selector = rasp.Select(rasp.tokens, rasp.tokens, rasp.Comparison.TRUE)
  return rasp.SelectorWidth(all_true_selector)

length = make_length()  # `length` is not a primitive in our implementation.
opp_index = length - rasp.indices - 1
flip = rasp.Select(rasp.indices, opp_index, rasp.Comparison.EQ)
reverse = rasp.Aggregate(flip, rasp.tokens)
```

Unfortunately, the only way to fix this is by having Tracr output the unembed matrix in its model params. I've tried to fix the demo without this, but we don't have enough information in Tracr's output (i.e., no way to know which part of the residual stream is the output space). I'll try to add the change in Tracr and then come back for this.

**Code example**

```python
def make_length():
  all_true_selector = rasp.Select(rasp.tokens, rasp.tokens, rasp.Comparison.TRUE)
  return rasp.SelectorWidth(all_true_selector)

length = make_length() 
opp_index = length - rasp.indices - 1
flip = rasp.Select(rasp.indices, opp_index, rasp.Comparison.EQ)
reverse = rasp.Aggregate(flip, rasp.tokens).named("reverse") # HERE IS THE CHANGE

n_heads = model.model_config.num_heads
n_layers = model.model_config.num_layers
d_head = model.model_config.key_size
d_mlp = model.model_config.mlp_hidden_size
act_fn = "relu"
normalization_type = "LN"  if model.model_config.layer_norm else None
attention_type = "causal"  if model.model_config.causal else "bidirectional"

n_ctx = model.params["pos_embed"]['embeddings'].shape[0]
d_vocab = model.params["token_embed"]['embeddings'].shape[0]
d_model = model.params["token_embed"]['embeddings'].shape[1]
d_vocab_out = model.params["token_embed"]['embeddings'].shape[0] - 2

cfg = HookedTransformerConfig(
    n_layers=n_layers,
    d_model=d_model,
    d_head=d_head,
    n_ctx=n_ctx,
    d_vocab=d_vocab,
    d_vocab_out=d_vocab_out,
    d_mlp=d_mlp,
    n_heads=n_heads,
    act_fn=act_fn,
    attention_dir=attention_type,
    normalization_type=normalization_type,
)
tl_model = HookedTransformer(cfg)

sd = {}
sd["pos_embed.W_pos"] = model.params["pos_embed"]['embeddings']
sd["embed.W_E"] = model.params["token_embed"]['embeddings']

# WRONG ASSUMPTION:
# The unembed is just a projection onto the first few elements of the residual stream, these store output tokens
# This is a NumPy array, the rest are Jax Arrays, but w/e it's fine.
sd["unembed.W_U"] = np.eye(d_model, d_vocab_out)

for l in range(n_layers):
    sd[f"blocks.{l}.attn.W_K"] = einops.rearrange(
        model.params[f"transformer/layer_{l}/attn/key"]["w"],
        "d_model (n_heads d_head) -> n_heads d_model d_head",
        d_head = d_head,
        n_heads = n_heads
    )
    sd[f"blocks.{l}.attn.b_K"] = einops.rearrange(
        model.params[f"transformer/layer_{l}/attn/key"]["b"],
        "(n_heads d_head) -> n_heads d_head",
        d_head = d_head,
        n_heads = n_heads
    )
    sd[f"blocks.{l}.attn.W_Q"] = einops.rearrange(
        model.params[f"transformer/layer_{l}/attn/query"]["w"],
        "d_model (n_heads d_head) -> n_heads d_model d_head",
        d_head = d_head,
        n_heads = n_heads
    )
    sd[f"blocks.{l}.attn.b_Q"] = einops.rearrange(
        model.params[f"transformer/layer_{l}/attn/query"]["b"],
        "(n_heads d_head) -> n_heads d_head",
        d_head = d_head,
        n_heads = n_heads
    )
    sd[f"blocks.{l}.attn.W_V"] = einops.rearrange(
        model.params[f"transformer/layer_{l}/attn/value"]["w"],
        "d_model (n_heads d_head) -> n_heads d_model d_head",
        d_head = d_head,
        n_heads = n_heads
    )
    sd[f"blocks.{l}.attn.b_V"] = einops.rearrange(
        model.params[f"transformer/layer_{l}/attn/value"]["b"],
        "(n_heads d_head) -> n_heads d_head",
        d_head = d_head,
        n_heads = n_heads
    )
    sd[f"blocks.{l}.attn.W_O"] = einops.rearrange(
        model.params[f"transformer/layer_{l}/attn/linear"]["w"],
        "(n_heads d_head) d_model -> n_heads d_head d_model",
        d_head = d_head,
        n_heads = n_heads
    )
    sd[f"blocks.{l}.attn.b_O"] = model.params[f"transformer/layer_{l}/attn/linear"]["b"]

    sd[f"blocks.{l}.mlp.W_in"] = model.params[f"transformer/layer_{l}/mlp/linear_1"]["w"]
    sd[f"blocks.{l}.mlp.b_in"] = model.params[f"transformer/layer_{l}/mlp/linear_1"]["b"]
    sd[f"blocks.{l}.mlp.W_out"] = model.params[f"transformer/layer_{l}/mlp/linear_2"]["w"]
    sd[f"blocks.{l}.mlp.b_out"] = model.params[f"transformer/layer_{l}/mlp/linear_2"]["b"]

for k, v in sd.items():
    sd[k] = torch.tensor(np.array(v))

tl_model.load_state_dict(sd, strict=False)

INPUT_ENCODER = model.input_encoder
OUTPUT_ENCODER = model.output_encoder

def create_model_input(input, input_encoder=INPUT_ENCODER):
    encoding = input_encoder.encode(input)
    return torch.tensor(encoding).unsqueeze(dim=0)

def decode_model_output(logits, output_encoder=OUTPUT_ENCODER, bos_token=INPUT_ENCODER.bos_token):
    max_output_indices = logits.squeeze(dim=0).argmax(dim=-1)
    decoded_output = output_encoder.decode(max_output_indices.tolist())
    decoded_output_with_bos = [bos_token] + decoded_output[1:]
    return decoded_output_with_bos

input = [bos, 1, 2, 3]
out = model.apply(input)
print("Original Decoding:", out.decoded)

input_tokens_tensor = create_model_input(input)
logits = tl_model(input_tokens_tensor)
decoded_output = decode_model_output(logits)
print("TransformerLens Replicated Decoding:", decoded_output)
```

Will output:
```
Original Decoding: ['BOS', 3, 2, 1]
TransformerLens Replicated Decoding: ['BOS', 1, 2, 3]
```

**System Info**
 * `transformer_lens` was installed using pip.
 * Linux.
 * Python 3.10.

**Additional context**

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/neelnanda-io/TransformerLens/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by abhinav-bellapu**
# Description

Fixes #481.

Ports the Tracr conversion demo from legacy `HookedTransformer` construction to `TransformerBridge.boot_native`. The PR adds a small optional Tracr conversion utility that reconstructs the categorical unembed from Tracr residual labels plus the output encoder, so named final RASP expressions no longer depend on occupying the first residual coordinates. It also teaches the native Bridge path to represent Tracr-shaped models by supporting `normalization_type=None` and respecting bidirectional attention.

No new runtime dependency is added; the Tracr helper is duck-typed and only needs a caller-supplied Tracr assembled model.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings in the checks run locally
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have not rewritten tests relating to key interfaces which would affect backward compatibility

## Validation

- `python3 -m py_compile transformer_lens/utilities/tracr.py transformer_lens/model_bridge/sources/native/model.py transformer_lens/model_bridge/supported_architectures/native.py transformer_lens/config/transformer_bridge_config.py tests/unit/test_tracr_conversion.py tests/unit/model_bridge/test_native_features.py`
- `python3 -m json.tool demos/Tracr_to_Transformer_Lens_Demo.ipynb >/dev/null`
- `awk 'length($0) > 100 { print FILENAME ":" FNR ":" length($0) ":" $0 }' transformer_lens/utilities/tracr.py transformer_lens/model_bridge/sources/native/model.py transformer_lens/model_bridge/supported_architectures/native.py transformer_lens/config/transformer_bridge_config.py tests/unit/test_tracr_conversion.py tests/unit/model_bridge/test_native_features.py`\n- `git diff --cached --check`\n\nNot run locally: pytest/nbval. This fresh temp clone had no Torch environment, and `uv run --no-sync` confirmed Torch was unavailable; a full dependency sync would be a large download and was avoided for this focused PR.

### CODE REVIEW

**Comment by abhinav-bellapu** [comment]
Superseded by #1460 from the clean branch.
